src/helpers/index.ts

```diff
@@ -0,0 +1,40 @@
+import { ethers } from "ethers";
+
+export function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export const chunkArray = <T>(array: T[], chunk_size: number) =>
+  Array(Math.ceil(array.length / chunk_size))
+    .fill(null)
+    .map((_, index) => index * chunk_size)
+    .map((begin) => array.slice(begin, begin + chunk_size));
+
+export async function chunkedPromiseAll<T>(
+  promises: Promise<T>[],
+  chunkSize = 100,
+  delayTime = 0
+) {
+  const chunks = chunkArray(promises, chunkSize);
+  const result: T[][] = [];
+
+  for (const chunk of chunks) {
+    if (result.length != 0) {
+      await delay(delayTime);
+    }
+    result.push(await Promise.all(chunk));
+  }
+
+  return result.flat();
+}
+
+export function padZeros(hex: string) {
+  if (hex.length == 66)
```

Comment:
small nit is that it would be nicer to have `{ }` on this if, instead of having to rely on the fact that someone reading knows it will execute the next line